### PR TITLE
Move sync block scan

### PR DESCRIPTION
**Pull request by PeterSolMS**
This PR implements some primitive work stealing by executing GCScan::GcWeakPtrScanBySingleThread on the first thread that finishes GCScan::GcWeakPtrScan and sort_mark_list.

### CODE REVIEW

File: src/coreclr/src/gc/gc.cpp [hunk 1]
```diff
@@ -21013,6 +21015,12 @@ void gc_heap::mark_phase (int condemned_gen_number, BOOL mark_only_p)
 //    unsigned long start = GetCycleCount32();
     sort_mark_list();
 //    printf("sort_mark_list took %u cycles\n", GetCycleCount32() - start);
+    // first thread to finish sorting will scan the sync syncblk cache
+    if (syncblock_scan_p == 0 && Interlocked::Increment(&syncblock_scan_p) == 1)
```
Comment (Maoni0) [inline src/coreclr/src/gc/gc.cpp:21019]: nit -

```
 if ((syncblock_scan_p == 0) && (Interlocked::Increment (&syncblock_scan_p) == 1))
```

**Review by Maoni0 — APPROVED** [review]
other than the nit, it LGTM

File: src/coreclr/src/gc/gc.cpp (hunk 1 above)
Comment (PeterSolMS) [reply src/coreclr/src/gc/gc.cpp:21019]: Done.

**Comment by Maoni0** [comment]
@PeterSolMS is on vacation and he didn't want to merge this till he gets back so please do not merge.

**Pull request merged**